src/Carbon/Carbon.php

```diff
@@ -2681,7 +2757,7 @@ public function diffFiltered(CarbonInterval $ci, Closure $callback, Carbon $dt =
 
         $diff = count($vals);
 
-        return $inverse && !$abs ? -$diff : $diff;
+        return $inverse && $abs === false ? -$diff : $diff;
```

Comment:
This could cause regressions, for example if someone pass preg_match result to $abs or anything that can be falsy but not exactly false.